crates/ruff_notebook/src/notebook.rs

```diff
@@ -410,11 +410,13 @@ impl Notebook {
 
     /// Return `true` if the notebook is a Python notebook, `false` otherwise.
     pub fn is_python_notebook(&self) -> bool {
-        self.raw
-            .metadata
-            .language_info
-            .as_ref()
-            .map_or(true, |language| language.name == "python")
+        if let Some(language_info) = self.raw.metadata.language_info.as_ref() {
+            return language_info.name == "python";
+        }
+        if let Some(kernel_spec) = self.raw.metadata.kernelspec.as_ref() {
+            return kernel_spec.language.as_deref() == Some("python");
+        }
+        false
```

Comment:
I'm not exactly sure why was the default value `true` in case `language_info` is absent but that doesn't seem correct to me. If we can't infer the language from either `language_info` or `kernelspec`, we'll now return `false` instead.

Reply:
Ok, so the notebooks in the ecosystem diff doesn't have both `language_info` and `kernelspec` in which case we're not marking it as Python notebook. I think I'd need to revert this change to return `true` in this case.